stdlib/src/utils/static_tuple.mojo

```diff
@@ -240,3 +240,162 @@ struct StaticTuple[element_type: AnyRegType, size: Int](Sized):
         )
         Pointer(ptr).store(val)
         self = tmp
+
+
+# ===----------------------------------------------------------------------===#
+# Array
+# ===----------------------------------------------------------------------===#
+
+
+@value
+struct InlineArray[ElementType: CollectionElement, size: Int](Sized):
+    """A fixed-size sequence of size homogenous elements where size is a constant expression.
+
+    Parameters:
+        ElementType: The type of the elements in the array.
+        size: The size of the array.
+    """
+
+    alias type = __mlir_type[
+        `!pop.array<`, size.value, `, `, Self.ElementType, `>`
+    ]
+    var _array: Self.type
+    """The underlying storage for the array."""
+
+    # This constructor will always cause a compile time error if used.
+    # It is used to steer users away from uninitialized memory.
+    @always_inline
+    fn __init__(inout self):
```

Comment:
I want to flag this constructor, which I'm using as a way to enforce safety guarantees. What do you think about this API? @JoeLoser @ConnorGray

Reply:
This makes sense, since otherwise a zero-argument call like so:

```
var arr = InlinedArray[Int, 10]()
```

would call the variadic constructor.  Doing it as an explicit constructor rather than a `@parameter if` in the variadic constructor makes sense for now.  I don't have a good gauge as to the compile time cost difference between the additional overload (constructor) vs. a `@parameter if` in the variadic constructor case.  Nothing to worry about for now as we're not at the level of compile time perf problems right now 😄

Reply:
FYI this constructor will need public doc strings since it's a public entity still.  We have a PR in-flight internally that adds a decorator (such as `@doc_private`) that we could apply to things like this in the future.  The API doc string checker wouldn't apply in such cases, and the constructor decl would be hidden in our generated docs too.

I moved your comment from above the constructor to be the doc string internally when I just imported this PR.  We also have a [PR in-flight](https://github.com/modularml/mojo/pull/2344) in the public `mojo` repo so the external CI here can do API doc string validation.